packages/tabs/src/tabs.vue

```diff
@@ -153,19 +154,19 @@ export default defineComponent({
     }
 
     const setCurrentName = value => {
-      if(currentName.value !== value && props.beforeLeave) {
-        const before = props.beforeLeave(value, currentName.value)
-        if(before && (before as Promise<void>).then) {
-          (before as Promise<void>).then(() => {
-            changeCurrentName(value)
-            nav$.value && nav$.value.removeFocus()
-          }, () => {
-            // ignore promise rejection in `before-leave` hook
-          })
-        } else if(before !== false) {
+      // should do nothing.
+      if (currentName.value === value) return
+
+      const beforeLeave = props.beforeLeave
+      const before = beforeLeave && beforeLeave(value, currentName.value)
+      if (before && isPromise(before)) {
+        (before as Promise<void>).then(() => {
```

Comment:
When `isPromise(before)` returns true, I don't think that you need to do this type casting yourself then, even thought there was nothing wrong with the code, could you check this out for me?

Reply:
This is the old code, I am modifying here, I also found the problem here

Reply:
done. @JeremyWuuuuu